### PR TITLE
add sync to specific projects or listen only

### PR DESCRIPTION
**Pull request by Michaelredaa**
## Changelog Description
Extend kitsu sync service with additional arguments to sync specific projects.

### CODE REVIEW

**Comment by EmberLightVFX** [comment]
Wouldn't it be better to have this as options in the UI? So an admin can select from a list what projects they want to sync?

**Comment by Michaelredaa** [comment]
It's a good idea specially when make control through OP UI only instead CLI. I extend the command to follow the manner that already exists to get qyick result.

File: openpype/modules/kitsu/kitsu_module.py [hunk 1]
```diff
@@ -124,19 +124,44 @@ def push_to_zou(login, password):
...
 
     Args:
         login (str): Kitsu user login
         password (str): Kitsu user password
+        projects (str): specific kitsu projects
+
+    SYNC_PROJECTS:
+    *: all projects
+    ^: dont sync any project just listen
+    "project01 project02 ...": to choose custom projects
```
Comment (Tilix4) [inline openpype/modules/kitsu/kitsu_module.py:156]: This `""` syntax is not consistent with others CLI and how `click` behaves. [Multiple option](https://click.palletsprojects.com/en/8.1.x/options/#multiple-options) must be used. I recommend a `-prj` short name option.

**Review by Tilix4 — CHANGES_REQUESTED** [review]
Nice feature! Thank you very much 🥇 

Left some suggestions and your linting is very strange. I suggest you to run `black --line-length 79 /path/to/my_script.py`.

Also, the documentation in `website/docs/module_kitsu.md` is missing, please write it.

File: openpype/modules/kitsu/utils/update_op_with_zou.py [hunk 2]
```diff
@@ -381,7 +383,26 @@ def sync_all_projects(
     dbcon = AvalonMongoDB()
     dbcon.install()
     all_projects = gazu.project.all_projects()
-    for project in all_projects:
+
+    project_to_sync = []
+    if specific_projects == ['*']:
+        project_to_sync = all_projects
```
[the remark this answers is not in the document]
Comment (Michaelredaa) [reply openpype/modules/kitsu/utils/update_op_with_zou.py]: I think we should implement it to support listen only, sync specific projects then listen and sync all projects then listen.

[the remark this answers is not in the document]
Comment (Tilix4) [reply openpype/modules/kitsu/utils/update_op_with_zou.py]: Okay, I agree with the feature, but I think it must be a dedicated argument and flag `--listen-only` to keep an explicit default behaviour.

[the remark this answers is not in the document]
Comment (EmberLightVFX) [reply openpype/modules/kitsu/utils/update_op_with_zou.py]: I would rather implement a listener argument so when this PR get merge the syncing for people that don't know about this update won't just stop syncing.

Also, what's the use case to only listen but not do anything with the result?

[the remark this answers is not in the document]
Comment (Michaelredaa) [reply openpype/modules/kitsu/utils/update_op_with_zou.py]: I have been updated code and docs with projects and listen only flags.
The main reason for listen only is the syncing process take a lot of time to sync all shots, assets, tasks, ..  so in some satiations we need to quick start listening to kitsu events without looping in all project data.

**Review by Tilix4 — CHANGES_REQUESTED** [review]
Thanks for the changes! Added new ones :smile_cat: 

PS: you still got linting issue, I recommend you to run `black` as explained in my first comment, it'll be easier for us to review.

File: website/docs/module_kitsu.md [hunk 3]
```diff
@@ -18,9 +18,20 @@ This setting is available for all the users of the OpenPype instance.
 ## Synchronize
 Updating OP with Kitsu data is executed running the `sync-service`, which requires to provide your Kitsu credentials with `-l, --login` and `-p, --password` or by setting the environment variables `KITSU_LOGIN` and `KITSU_PWD`. This process will request data from Kitsu and create/delete/update OP assets.
 Once this sync is done, the thread will automatically start a loop to listen to Kitsu events.
+- `-prj, --project` This flag accepts multiple project name to sync specific projects, and the default to sync all projects.
+- `-lo, --listen-only` This flag to run listen to Kitsu events only without any sync.
+
+Note: You must use one argument of `-pro` or `-lo`, because the listen only flag override syncing flag.
```
[the remark this answers is not in the document]
Comment (Michaelredaa) [reply website/docs/module_kitsu.md:24]: I think the kitsu events are for all active projects, if we customize events for filtered projects this consider as an extra layer in OP syncing. As well as if we run listening for all project it not costy and make sense more to detect any changes for all projects onece.

[the remark this answers is not in the document]
Comment (Tilix4) [reply website/docs/module_kitsu.md:24]: I think we'll face the use case I describe sooner than we think. But we can keep it like that until then.

File: website/docs/module_kitsu.md [hunk 4]
```diff
@@ -18,9 +18,20 @@ This setting is available for all the users of the OpenPype instance.
...
+Note: You must use one argument of `-pro` or `-lo`, because the listen only flag override syncing flag. 
 
 ```bash
+// sync all projects then run listen
 openpype_console module kitsu sync-service -l me@domain.ext -p my_password
+
+// sync specific projects then run listen
+openpype_console module kitsu sync-service -l me@domain.ext -p my_password -prj project_name01 -prj  project_name02
+
+// start listen only
```
Comment (Tilix4) [inline website/docs/module_kitsu.md:33]: If we don't implement filtering for listening, I'd be more specific

```suggestion
// start listen only for all projects
```

**Review by Tilix4 — CHANGES_REQUESTED** [review]
Nice, we are almost there, still one code comment missing and added a suggestion

File: openpype/modules/kitsu/kitsu_module.py [hunk 5]
```diff
@@ -128,15 +128,35 @@ def push_to_zou(login, password):
 @click.option(
     "-p", "--password", envvar="KITSU_PWD", help="Password for kitsu username"
 )
-def sync_service(login, password):
+@click.option(
+    "-prj",
```
Comment (Minkiu) [inline openpype/modules/kitsu/kitsu_module.py:132]: Shouldn't these be:
```py
"-prjs",
"--projects"
```

Comment (Michaelredaa) [reply openpype/modules/kitsu/kitsu_module.py:132]: The arg `--project` accept one input parameter not many parameters

Comment (Minkiu) [reply openpype/modules/kitsu/kitsu_module.py:132]: Ah I see, the last "projects" in line `134` would collect any multiple `-prj` provided?

**Review by Tilix4 — APPROVED** [review]
Nice! LGTM!

**Review by mkolar — APPROVED** [review]

**Pull request merged**